src/ray/core_worker/reference_count.cc

```diff
@@ -492,29 +496,47 @@ void ReferenceCounter::DeleteReferenceInternal(ReferenceTable::iterator it,
         DeleteReferenceInternal(inner_it, deleted);
       }
     }
-  }
-
-  // Perform the deletion.
-  if (should_delete_value) {
+    // Perform the deletion.
     ReleasePlasmaObject(it);
     if (deleted) {
       deleted->push_back(id);
     }
+
+    if (it->second.reconstructable_owned_objects_it.has_value()) {
```

Comment:
Isn't it possible to include this logic to `ReleaseLineageReferences `? (Not 100% sure though..)

Reply:
I don't think so, but probably we can also do this in the new EraseReference call.